### PR TITLE
Feature/connect login to backend

### PR DESCRIPTION
**Pull request by fnoori**
User is able to login

### CODE REVIEW

**Comment by adnanrh** [comment]
Looks good. The dialog just closes though if the credentials are incorrect

**Comment by fnoori** [comment]
Yeah that's one thing I couldn't quite figure out.

The way this dialog was originally implemented, it makes it difficult to check for a fail case when logging in. I'll do some digging to see if I can get anywhere with it, if I can't find anything, I'll just have it so that it stays upon. This will be an absolute last resort.

File: src/store/auth.js [hunk 1]
```diff
@@ -39,8 +41,13 @@ export default {
     socket_updateUserData(/* state */) {
       console.log('socket_updateUserData');
     },
-    socket_login(/* state */) {
-      console.log('socket_login');
+    socket_login(state, token) {
```
Comment (nxtman123) [inline src/store/auth.js]: Use the "outside a vue file" syntax from https://v1.quasar-framework.org/quasar-plugins/notify#Notify-API here to notify with the response's `value` if `!isSuccessful.

File: src/layouts/MyLayout.vue [hunk 2]
```diff
@@ -170,6 +170,15 @@ export default {
...
   },
+  watch: {
+    loggedIn(newValue, oldValue) {
+      if (oldValue === false && newValue === true) {
+        this.$q.notify('Logged in successfully');
+      } else if (oldValue === false && newValue === false) {
+        this.$q.notify('Failed to login');
+      }
+    },
+  },
```
Comment (nxtman123) [inline src/layouts/MyLayout.vue:179]: This won't be necessary (see below)

File: src/components/LoginDialog.vue [hunk 3]
```diff
@@ -75,9 +75,10 @@ export default {
   methods: {
     onSubmit() {
       if (this.readyToSubmit) {
-        // TODO: replace alert with call to server to authenticate
-        // eslint-disable-next-line no-alert
-        alert(`TODO: call server to log in ${this.username}`);
+        this.$socket.emit('login', {
+          username: this.username,
+          password: this.password,
+        });
```
Comment (nxtman123) [inline src/components/LoginDialog.vue:80]: :ok_hand:

File: src/store/auth.js [hunk 4]
```diff
@@ -39,8 +41,13 @@ export default {
...
+      try {
+        LocalStorage.set('token', token);
```
Comment (nxtman123) [inline src/store/auth.js]:
```suggestion
        LocalStorage.set('token', token.value);
```

**Review by nxtman123 — CHANGES_REQUESTED** [review]

File: src/store/auth.js [hunk 5]
```diff
@@ -39,8 +41,15 @@ export default {
...
     },
-    socket_login(/* state */) {
-      console.log('socket_login');
+    socket_login(state, token) {
+      try {
+        LocalStorage.set('token', token.value);
+        state.user = jwt.decode(token.value, { complete: true }).payload;
+        Notify.create('Logged in successfully');
+      } catch (e) {
+        Notify.create('Failed to login');
```
Comment (nxtman123) [inline src/store/auth.js]: you should test `isSuccessful` and if it's not, notify with the server's message
```suggestion
        Notify.create(token.value);
```

also, I recommend changing `token` to `loginResult` to match the object's name on the backend

**Review by nxtman123 — CHANGES_REQUESTED** [review]

**Review by nxtman123 — APPROVED** [review]

**Pull request merged**